### PR TITLE
chore: update contributing file

### PR DESCRIPTION
**Pull request by Ehesp**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,61 +1,118 @@
...
+
+```bash
+melos bootstrap
+```
+
+The bootstrap command locally links all dependencies within the project without having to
+provide manual [`dependency_overrides`](https://dart.dev/tools/pub/pubspec). This allows all
+plugins, examples and tests to build from the local clone project.
+
+## 4. Running an example
```
Comment (kroikie) [inline CONTRIBUTING.md:53]: Could we include a note here that running `melos bootstrap` is required before running the examples?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -67,37 +124,45 @@ keep the code consistent and avoid common pitfalls.
...
   "Compare & pull request" button
 
 Please make sure all your checkins have detailed commit messages explaining the patch.
 
-For pull requests that affect only one Flutterfire plugin, use a title that starts
-with the name of the plugin in brackets (e.g. [cloud_firestore]).
+When naming the title of your pull request, please follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
+guide. For example, for a fix to the Cloud Firestore plugin:
+
+`fix(cloud_firestore): Fixed a bug!`
```
Comment (Salakar) [inline CONTRIBUTING.md:158]:
```suggestion
`fix(cloud_firestore): fixed a bug!`
```

Also could we use a different plugin in the example given the potential upcoming rename of cloud_firestore, one less thing to rename later or not forget about?

File: CONTRIBUTING.md (hunk 1 above)
Comment (Salakar) [reply CONTRIBUTING.md:53]: Mentioned in the previous step above, though I think it's worth maybe adding an additional bit of info that mentions that `pub get` inside any package in the project is not required once the project has been bootstrapped and that if dependencies have been changed you just need to run bootstrap again

Comment (kroikie) [reply CONTRIBUTING.md:53]: Sounds good, lets mention that!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,61 +1,118 @@
...
+cd packages/cloud_firestore/cloud_firestore/example
+flutter drive --target=./test_driver/cloud_firestrore_e2e.dart
 ```
 
-To run the integration tests:
+To run tests against web environments, run the command as a release build:
 
+```bash
+cd packages/cloud_firestore/cloud_firestore/example
+flutter drive --target=./test_driver/cloud_firestrore_e2e.dart --release -d chrome
```
Comment (Salakar) [inline CONTRIBUTING.md:99]:
```suggestion
flutter drive --target=./test_driver/cloud_firestore_e2e.dart --release -d chrome
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,61 +1,118 @@
...
+
+### End-to-end (e2e) tests
+
+E2e tests are those which directly communicate with Firebase, whose results cannot be mocked. These tests run directly from
+an example application. To run e2e tests, run the `flutter drive` command from the plugins main `example` directory, targeting the
+entry e2e test file:
+
+```bash
+cd packages/cloud_firestore/cloud_firestore/example
+flutter drive --target=./test_driver/cloud_firestrore_e2e.dart
```
Comment (Salakar) [inline CONTRIBUTING.md:92]:
```suggestion
flutter drive --target=./test_driver/cloud_firestore_e2e.dart
```

**Comment by dackers86** [comment]
@googlebot I consent

**Pull request merged**